examples/hnas/example_mlp.py

```diff
@@ -0,0 +1,228 @@
+        correct = 0
+        total = 0
+        for data, target in valid_loader:
+            output = model(data.view(batch_size, -1))
+            _, predicted = torch.max(output.data, 1)
+            total += target.size(0)
+            correct += (predicted == target).sum().item()
+
+        acc = correct / total
+
+        # Every time we improve in terms of validation accuracy, we save the network parameters.
+        if curr_best is None or acc > curr_best:
+            curr_best = acc
+            print("checkpoint model")
+            torch.save(model.state_dict(), 'model_checkpoint')
+
+        # We also report the hyperparameter configuration back to AutoGluon,
+        # so we can process it later in the callback function, for example for visualization.
+        config = dict()
+        config["n_units_1"] = n_units_1
+        config["n_units_2"] = n_units_2
+        config["dropout_1"] = dropout_1
+        config["dropout_2"] = dropout_2
+        config["batch_size"] = batch_size
+        config["learning_rate"] = learning_rate
+        config["wd"] = wd
+
+        ts_now = time.time()
+        eval_time = ts_start - ts_now
+        reporter(
+            epoch=epoch + 1,
+            performance=float(curr_best),
+            eval_time=eval_time,
+            time_step=ts_now, **config)
+
+
+def callback(training_history, start_timestamp):
+    # This callback function will be executed every time AutoGluon collected some new data.
+    # In this example we will parse the training history into a .csv file and save it to disk, such that we can
+    # for example plot AutoGluon's performance during the optimization process.
+    # If you don't care about analyzing performance of AutoGluon online, you can also ignore this callback function and
+    # just save the training history after AutoGluon has finished.
+    import pandas as pd
+    task_dfs = []
+
+    # this function need to be changed if you return something else than accuracy
+    def compute_error(df):
+        return 1.0 - df["performance"]
+
+    def compute_runtime(df, start_timestamp):
+        return df["time_step"] - start_timestamp
+
+    for task_id in training_history:
+        task_df = pd.DataFrame(training_history[task_id])
+        task_df = task_df.assign(task_id=task_id,
+                                 runtime=compute_runtime(task_df, start_timestamp),
+                                 error=compute_error(task_df),
+                                 target_epoch=task_df["epoch"].iloc[-1])
+        task_dfs.append(task_df)
+
+    result = pd.concat(task_dfs, axis="index", ignore_index=True, sort=True)
+
+    # re-order by runtime
+    result = result.sort_values(by="runtime")
+
+    # calculate incumbent best -- the cumulative minimum of the error.
+    result = result.assign(best=result["error"].cummin())
+
+    result.to_csv("history.csv")
+
+
+# CLI
+def parse_args():
+    parser = argparse.ArgumentParser(description='runs autogluon on autoff benchmarks')
```

Comment:
Don't think users will understand what `autoff benchmarks` is. May want to describe it more or provide external link to more info.

Reply:
good catch, was an artifact